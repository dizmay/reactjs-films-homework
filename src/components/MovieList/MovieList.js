/* eslint-disable react/no-array-index-key */
import React from 'react';
import PropTypes from 'prop-types';
import MovieCard from '../MovieCard/MovieCard';
import styles from './MovieList.scss';

const MovieList = ({
  popularMovies, genres, movieTrailer, getMovieTrailer,
}) => (
  <div className={styles.movieList__container}>
    {
      popularMovies.length > 0
        ? (
          <div className={styles.movieList__grid}>
            {
              popularMovies.map((movie, index) => (
                <MovieCard
                  key={index}
                  title={movie.title}
                  rate={movie.vote_average}
                  poster={movie.poster_path}
                  movieGenres={movie.genre_ids}
                  genres={genres}
                  overview={movie.overview}
                  movieId={movie.id}
                  movieTrailer={movieTrailer}
                  getMovieTrailer={getMovieTrailer}
                />
              ))
            }
          </div>
        )
        : <h5 className={styles.movieList__empty}>No results</h5>
    }
  </div>
);

MovieList.defaultProps = {
  popularMovies: [],
  genres: [],
  movieTrailer: '',
  getMovieTrailer: null,
};

MovieList.propTypes = {
  popularMovies: PropTypes.arrayOf(PropTypes.object),
  genres: PropTypes.arrayOf(PropTypes.object),
  movieTrailer: PropTypes.string,
  getMovieTrailer: PropTypes.func,
};

export default MovieList;
