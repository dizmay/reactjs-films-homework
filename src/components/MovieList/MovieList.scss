* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.movieList__container {
  width: 100vw;
  min-height: 20vh;
  background: #e6e6e6e6;
  display: flex;
  justify-content: center;
  padding: 2rem 0;
}

.movieList__grid {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  gap: 2rem;
  align-items: center;
  justify-items: center;
}

.movieList__empty {
  font-size: 5rem;
}
