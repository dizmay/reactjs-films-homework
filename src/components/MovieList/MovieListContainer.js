/* eslint-disable react/no-array-index-key */
import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { fetchGenres, fetchTopRated, fetchMovieTrailer } from '../../modules/Search/searchActions';
import MovieList from './MovieList';

const MovieListContainer = ({
  popularMovies,
  getTopRated,
  getGenres,
  genres,
  movieTrailer,
  getMovieTrailer,
}) => {
  useEffect(() => {
    getTopRated();
    getGenres();
  }, []);
  return (
    <MovieList
      popularMovies={popularMovies}
      genres={genres}
      movieTrailer={movieTrailer}
      getMovieTrailer={getMovieTrailer}
    />
  );
};

MovieListContainer.defaultProps = {
  popularMovies: [],
  getTopRated: null,
  getGenres: null,
  genres: [],
  movieTrailer: '',
  getMovieTrailer: null,

};

MovieListContainer.propTypes = {
  popularMovies: PropTypes.arrayOf(PropTypes.object),
  getTopRated: PropTypes.func,
  getGenres: PropTypes.func,
  genres: PropTypes.arrayOf(PropTypes.object),
  movieTrailer: PropTypes.string,
  getMovieTrailer: PropTypes.func,
};

const mapStateToProps = (state) => ({
  popularMovies: state.search.movies,
  genres: state.search.genres,
  movieTrailer: state.search.movieTrailer,
});

const mapDispatchToProps = (dispatch) => ({
  getTopRated: () => dispatch(fetchTopRated()),
  getGenres: () => dispatch(fetchGenres()),
  getMovieTrailer: (movieId) => dispatch(fetchMovieTrailer(movieId)),
});

export default connect(mapStateToProps, mapDispatchToProps)(MovieListContainer);
