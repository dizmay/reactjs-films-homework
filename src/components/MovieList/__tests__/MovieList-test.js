import React from 'react';
import renderer from 'react-test-renderer';
import MovieList from '../MovieList';

it('renders correctly', () => {
  const tree = renderer.create(<MovieList />).toJSON();
  expect(tree).toMatchSnapshot();
});
