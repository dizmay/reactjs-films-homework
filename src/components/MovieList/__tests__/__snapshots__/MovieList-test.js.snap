// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="movieList__container"
>
  <h5
    className="movieList__empty"
  >
    No results
  </h5>
</div>
`;
